### PR TITLE
fix(ui-react): Whitespace in Tab IDs

### PR DESCRIPTION
**Pull request by axelEandrews**
Adds `groupId` to TabsContext such that unique IDs can be created for each tab group. IDs are of the form `${groupId}-tab-${value}` for tabs, and`${groupId}-panel-${value}` for panels. Also adds `whitespaceValue` to TabsContext, which is assigned a value in TabsContainer, then used to remove whitespace from `value` in TabsItem and TabsPanel (ensures proper operation regardless of how Tabs are created by user).

Outer html output for a tablist with given values: ['Tab 1', 'Tab 2', 'Tab 3']: `<div role="tablist" class="amplify-tabs__list" style="justify-content: flex-start;"><button role="tab" id="amplify-id-:rl:-tab-Tab-1" aria-selected="true" aria-controls="amplify-id-:rl:-panel-Tab-1" class="amplify-tabs__item amplify-tabs__item--active">Tab 1</button><button role="tab" id="amplify-id-:rl:-tab-Tab-2" aria-selected="false" aria-controls="amplify-id-:rl:-panel-Tab-2" tabindex="-1" class="amplify-tabs__item">Tab 2</button><button disabled="" role="tab" id="amplify-id-:rl:-tab-Tab-3" aria-selected="false" aria-controls="amplify-id-:rl:-panel-Tab-3" tabindex="-1" class="amplify-tabs__item">Disabled tab</button></div>`

#### Issue #, if available
#5220 

#### Description of how you validated changes
Added tests to confirm IDs are the same for tabs in the same tab group with the same given value (should be avoided in practice), IDs are different for tabs in different groups with the same value (fine to do in practice). Also confirms that tabs in same group with different values yield unique IDs. Passes all yarn tests.

#### Checklist

- [x] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [x] PR description included
- [] Relevant documentation is changed or added (and PR referenced)
- [x] `yarn test` passes and tests are updated/added
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react/src/primitives/Tabs/Tabs.tsx
Comment (reesscot) [inline packages/react/src/primitives/Tabs/Tabs.tsx:28]: qq: are there any items in rest that we don't want to spread into TabPanel? Geniunely asking as I don't know. I've seen cases in the past where we have passed down properties that aren't appropriate, which can throw a react warning in the console.

File: packages/react/src/primitives/Tabs/TabsContainer.tsx
Comment (reesscot) [inline packages/react/src/primitives/Tabs/TabsContainer.tsx:25]: Since this never changes, it doesn't need to be declared inside of the component and should be moved outside of the component (near the top of file, under imports)

File: packages/react/src/primitives/Tabs/Tabs.tsx
Comment (axelEandrews) [reply packages/react/src/primitives/Tabs/Tabs.tsx:28]: `rest` just contains `label: React.ReactNode` for the TabPanel, and for the TabList it just has `isDisabled: Boolean | undefined`. We may as well just pass them directly at that point if that's intended, right?

Comment (calebpollman) [reply packages/react/src/primitives/Tabs/Tabs.tsx:28]: If additional parameters are added to the parent interface they'll bleed through. Is that behavior what we want here?

File: packages/react/src/primitives/Tabs/TabsContext.tsx
Comment (calebpollman) [inline packages/react/src/primitives/Tabs/TabsContext.tsx:6]: Why is `groupId` optional?

File: packages/react/src/primitives/Tabs/TabsItem.tsx
Comment (calebpollman) [inline packages/react/src/primitives/Tabs/TabsItem.tsx:21]: Since `whitespaceValue` is a static value, there is no need to pass it through context as it will not update

File: packages/react/src/primitives/Tabs/TabsContainer.tsx
Comment (calebpollman) [inline packages/react/src/primitives/Tabs/TabsContainer.tsx:12]: Module scoped constants should be declared in SCREAMING_SNAKE_CASE

File: packages/react/src/primitives/Tabs/TabsContainer.tsx
Comment (calebpollman) [inline packages/react/src/primitives/Tabs/TabsContainer.tsx:26]: It would be helpful to add a small comment explaining the use case for `groupId` for future maintainers

File: packages/react/src/primitives/Tabs/Tabs.tsx
Comment (axelEandrews) [reply packages/react/src/primitives/Tabs/Tabs.tsx]: I removed this spread, wasn't needed so I think it's safer to just get rid of it.

File: packages/react/src/primitives/Tabs/TabsContainer.tsx
Comment (dbanksdesign) [reply packages/react/src/primitives/Tabs/TabsContainer.tsx]: Love the name SCREAMING_SNAKE_CASE

**Review by dbanksdesign — APPROVED** [review]
LGTM

**Review by hbuchel — COMMENTED** [review]
Just took a look at the Tabs docs page from this branch; it looks like somewhere we've created a breaking change because the Tabs no longer default to showing an open tab if the existing tab name had a space in it. This is what I'm seeing locally, can you please verify? 
Screenshot 2024-07-09 at 3 02 39 PM

**Review by hbuchel — CHANGES_REQUESTED** [review]
Just throwing a block on this for now until the breaking change is addressed. Also, I re-ran the Docs e2e test (not sure why at first glance it failed)

**Pull request closed without merging**
